### ISSUE TITLE
No matches for kind "GitRepository" in version "source.toolkit.fluxcd.io/v1beta2"

### ISSUE THREAD

**Issue by chrischen** [fluxcd/flux2#2645]
### Describe the bug

All of a sudden I'm getting
`error: no matches for kind "GitRepository" in version "source.toolkit.fluxcd.io/v1beta2"`

when running `flux reconcile kustomization flux-system`

Kustomizations seem to still be syncing from the git repo, but running that command gives the error in the title.

Inside the generated Kustomizations:
```
app.kubernetes.io/instance: flux-system
app.kubernetes.io/part-of: flux
app.kubernetes.io/version: v0.23.0
```

`flux version` Output
```
flux: v0.23.0
helm-controller: v0.13.0
image-automation-controller: v0.17.1
image-reflector-controller: v0.13.2
kustomize-controller: v0.18.0
notification-controller: v0.18.1
source-controller: v0.18.0
```

Kubectl client version v1.23.5.
Kubernetes version v1.19.6.

Full output
```
► annotating Kustomization flux-system in flux-system namespace
✔ Kustomization annotated
◎ waiting for Kustomization reconciliation
✗ Kustomization reconciliation failed: GitRepository/flux-system/flux-system dry-run failed, error: no matches for kind "GitRepository" in version "source.toolkit.fluxcd.io/v1beta2"
```
### Steps to reproduce

Update flux client from 0.23.0 to 0.28.5.
Run `flux reconcile kustomization flux-system`
This initially was working, and at some point running it again produces the error.

Tried on both flux version 0.28.5 and went back to flux version 0.23.0.

### Expected behavior

Reconciliation was working before (image updates and kustomization changes still seem to be working automatically—just not with this command). I didn't make any manual changes to flux  on the server (did not upgrade the server controllers) nor upgrade kubernetes. The only change I've done was upgrade flux client to 0.28.5 and run some commands.

### OS / Distro

macOS 12.3.1 arm

### Flux version

0.28.5 and 0.23.0

### Flux check

► checking prerequisites
✗ flux 0.23.0 <0.28.5 (new version is available, please upgrade)
✔ Kubernetes 1.19.6 >=1.19.0-0
► checking controllers
✔ helm-controller: deployment ready
► ghcr.io/fluxcd/helm-controller:v0.13.0
✔ image-automation-controller: deployment ready
► ghcr.io/fluxcd/image-automation-controller:v0.17.1
✔ image-reflector-controller: deployment ready
► ghcr.io/fluxcd/image-reflector-controller:v0.13.2
✔ kustomize-controller: deployment ready
► ghcr.io/fluxcd/kustomize-controller:v0.18.0
✔ notification-controller: deployment ready
► ghcr.io/fluxcd/notification-controller:v0.18.1
✔ source-controller: deployment ready
► ghcr.io/fluxcd/source-controller:v0.18.0
✔ all checks passed

(Running flux check on latest 0.28.5 reports my kubernetes version is out of date, but the problem appears with flux 0.23.0 as well).

### Git provider

GitLab

### Container Registry provider

DockerHub

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by hiddeco** [comment]
[Flux `v0.28.x` moved to the `v1beta2` Source API](https://github.com/fluxcd/flux2/releases/tag/v0.28.0). You either need to update your cluster to match this version, or downgrade the CLI version to `v0.27.x`. Multi-version support in the CLI is not planned until Flux reaches GA.

**Comment by chrischen** [comment]
Cool I confirmed this was the issue. Upgrading to 0.28 caused it and in an attempt to fix it I manually changed the GitRepository manifest to say `apiVersion: source.toolkit.fluxcd.io/v1beta2` which then seemed to cause the issue even when using < 0.27 of flux client. I reverted the manifest change to `apiVersion: source.toolkit.fluxcd.io/v1beta1` and used 0.27 of flux client and it resolved this issue. Thanks!
